tests/versioned/apollo-server-hapi/segments.tests.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+
+const utils = require('@newrelic/test-utilities')
+utils.assert.extendTap(tap)
+
+const { getTypeDefs, resolvers } = require('../data-definitions')
+const { createSegmentsTests } = require('./hapi-segments-tests')
+
+tap.test('apollo-server-hapi: segments', (t) => {
+  t.autoend()
+
+  let server = null
+  let hapiServer = null
+  let serverUrl = null
+  let helper = null
+  
+  t.beforeEach(async () => {
+    // load default instrumentation. hapi being critical
+    helper = utils.TestAgent.makeInstrumented()
+    const createPlugin = require('../../../lib/create-plugin')
+    const nrApi = helper.getAgentApi()
+
+    // TODO: eventually use proper function for instrumenting and not .shim
+    const plugin = createPlugin(nrApi.shim)
+
+    const Hapi = require('@hapi/hapi')
+    
+    const graphqlPath = '/gql'
+
+    // Do after instrumentation to ensure hapi isn't loaded too soon.
+    const { ApolloServer, gql } = require('apollo-server-hapi')
+    server = new ApolloServer({
+      typeDefs: getTypeDefs(gql),
+      resolvers,
+      plugins: [plugin]
+    })
+
+    hapiServer = Hapi.server({
+      host: 'localhost',
+      port: 5000
+    })
+
+    await server.applyMiddleware({ app: hapiServer, path: graphqlPath })
+
+    await server.installSubscriptionHandlers(hapiServer.listener)
+
+    await hapiServer.start()
+
+    serverUrl = `http://localhost:${hapiServer.settings.port}${graphqlPath}`
+    t.context.helper = helper
+    t.context.serverUrl = serverUrl
+  })
+
+  t.afterEach((done) => {
+    hapiServer && hapiServer.stop()
+    server && server.stop()
+
+    helper.unload()
+    server = null
+    serverUrl = null
+    helper = null
+
+    clearCachedModules(['@hapi/hapi', 'apollo-server-hapi'], () => {
+      done()
+    })
+  })
+
+  createSegmentsTests(t)
```

Comment:
Not sure the two-file approach is really appropriate here. In the other cases, there's a shared file reused between multiple tests (two different projects that use express in different ways, for example). In this case, there's only one usage so it just sort of obfuscates things / makes them slightly more complicated than necessary.

Tis minor though, don't have to resolve with this PR.

Reply:
Great point!  I'll fix that quick.